### PR TITLE
Changes target flags to pull more intuitively

### PR DESCRIPTION
**Pull request by KipHamiltons**

### CODE REVIEW

File: tools/target.py [hunk 1]
```diff
@@ -53,19 +51,31 @@ def run(target, build, should_pull, **kwargs):
...
-            build = True
-            should_pull = False
-
-        if should_pull:
+        if pull:
             # Pull the image from dockerhub
             platform.pull(defaults.image, target)
 
-        # If user wants or needs to build it locally, do that
+        # If user wants or needs to build it locally, do that unless the build context is the same as origin/master
```
Comment (abihall) [inline tools/target.py:58]: Where does it check if the build context is the same as origin/master? I can't see where it does that

File: tools/target.py [hunk 2]
```diff
@@ -21,21 +21,19 @@ def register(command):
         "--build",
         dest="build",
         action="store_true",
-        default=False,
         help="build the image locally, instead of pulling and tagging it",
     )
 
     command.add_argument(
```
Comment (Crafty15) [inline tools/target.py:27]: Nice comments down further. Would be super cool to have similar comments up here.

File: tools/target.py (hunk 1 above)
Comment (Bidski) [reply tools/target.py:58]: https://github.com/NUbots/NUbots/blob/a85369bf060cb27c54dfd4476c6e24e109214a48/tools/target.py#L45-L52

Comment (KipHamiltons) [reply tools/target.py:58]: just above that in the file (in parts we didn't change), we run `git diff` to check if there are changes in the `docker` directory. If there are, the "build context" has been changed (and is "dirty")

File: tools/target.py [hunk 3]
```diff
@@ -53,19 +51,31 @@ def run(target, build, should_pull, **kwargs):
...
+                cprint(
+                    "There aren't any changes to be made to the docker image. "
+                    "Skipping build, and using tagged image instead",
+                    "red",
+                    attrs=["bold"],
+                )
+            else:
+                platform.build(defaults.image, target)
+        # If the user hasn't selected build but they have changed the docker directory, then we warn them that the image
+        # isn't going to be built, even though the dirty diff implies they might want it to be built
```
Comment (Crafty15) [inline tools/target.py:70]: What is a diff? and why is a dirty one bad?

**Review by Bidski — APPROVED** [review]
Seems more logical to me (but then again this was my idea)

File: tools/target.py [hunk 4]
```diff
@@ -53,19 +51,31 @@ def run(target, build, should_pull, **kwargs):
...
+        else:
+            if changed:
```
Comment (Crafty15) [inline tools/target.py:72]: What is the construct that changed is initialised with (line 45). Not in your changes, just wondering.

**Review by Crafty15 — COMMENTED** [review]
Nice one bruv. Mind giving me a quick run down on what this does?

File: tools/target.py (hunk 1 above)
Comment (abihall) [reply tools/target.py:58]: Okay thanks I see it now

File: tools/target.py (hunk 3 above)
Comment (Y0sh1dk) [reply tools/target.py:70]: Dirty diff here is meaning that your local `docker` directory is different to the `docker` directory that is on `origin/master`. In this case you would usually want to rebuild a docker image with your local changes, if you decide not too, a warning is displayed.

File: tools/target.py (hunk 4 above)
Comment (Y0sh1dk) [reply tools/target.py:72]: `changed` gets set to `true` on line 45 if the git diff returns a exit code other than 0 (this happens when your local docker directory is different to the docker directory that is on origin/master

**Pull request merged**